modules/account-lib/src/coin/hbar/tokenTransferBuilder.ts

```diff
@@ -0,0 +1,104 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import * as Long from 'long';
+import { TokenId } from '@hashgraph/sdk';
+import { proto } from '../../../resources/hbar/protobuf/hedera';
+import { BuildTransactionError, InvalidParameterValueError } from '@bitgo/sdk-core';
+import { TransferBuilder } from './transferBuilder';
+import { Transaction } from './transaction';
+import { getHbarTokenFromTokenName, isValidToken, stringifyAccountId, stringifyTokenId } from './utils';
+
+export class TokenTransferBuilder extends TransferBuilder {
+  private _tokenId: string;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this._txBodyData.tokenTransfers = [this.buildTokenTransferData()]; // set to list by the contract
+    return await super.buildImplementation();
+  }
+
+  private buildTokenTransferData(): proto.ITokenTransferList {
+    return {
+      token: this.buildTokenId(this._tokenId),
+      transfers: [
+        { accountID: this.buildAccountData(this._source.address), amount: Long.fromString(this._amount).negate() }, // sender
+        { accountID: this.buildAccountData(this._toAddress), amount: Long.fromString(this._amount) }, // recipient
+      ],
+    };
+  }
+
+  private buildTokenId(tokenId: string): proto.TokenID {
+    const tokenData = TokenId.fromString(tokenId);
+    return new proto.TokenID({
+      tokenNum: tokenData.num,
+      realmNum: tokenData.realm,
+      shardNum: tokenData.shard,
+    });
+  }
+
+  /** @inheritdoc */
+  initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+    const transferData = tx.txBody.cryptoTransfer;
```

Comment:
Be sure to set the transaction type here, i.e., `this.transaction.setTransactionType(TransactionType.Send)`